smoke-tests/SmokeTestGenerator.js

```diff
@@ -45,43 +45,67 @@ function buildPath(pathSegs, delim) {
 function generateScriptBody(isBash, delim) {
 	const projectsPath = ['test', 'code-fixtures', 'projects'];
 	const customRulePath = ['test', 'test-jars', 'apex', 'testjar1.jar'];
+	const customConfigPath = ['test', 'test-xml', 'apex', 'smoke-config.xml'];
 	const resultsPath = ['smoke-test-results'];
 
 	const exeName = isBash ? '$EXE_NAME' : '%EXE_NAME%';
 
 	// Declare an array with all of the commands we intend to execute.
 	const commands = [
+		// Log a header.
 		`echo "====== STARTING SMOKE TEST ======"`,
+		// Create the results directory.
 		`echo "==== Make results directory ===="`,
 		isBash ? `mkdir -p smoke-test-results` : `if not exist smoke-test-results mkdir smoke-test-results || exit /b 1`,
+		// List all the rules w/out filters.
 		`echo "==== List all rules w/out filters ===="`,
 		`${exeName} scanner:rule:list`,
+		// List rules, filtered by engine.
 		`echo "==== Filter rules by engine ===="`,
 		`${exeName} scanner:rule:list --engine eslint`,
+		// Describe a real rule.
 		`echo "==== Describe a real rule ===="`,
 		`${exeName} scanner:rule:describe -n EmptyCatchBlock`,
+		// Describe a non-existent rule.
 		`echo "==== Describe a non-existent rule ===="`,
 		`${exeName} scanner:rule:describe -n NotAnActualRule`,
+		// Run such that PMD and ESLint are invoked.
 		`echo "==== Run rules against force-app, which should hit PMD and ESLint engines ===="`,
-		`${exeName} scanner:run --format junit --target ${buildPath([...projectsPath, 'app', 'force-app'], delim)} --outfile ${buildPath([...resultsPath, 'run1.xml'], delim)}`,
+		`${exeName} scanner:run --format junit --target ${buildPath([...projectsPath, 'app', 'force-app'], delim)} --outfile ${buildPath([...resultsPath, 'pmd-eslint.xml'], delim)}`,
```

Comment:
Changed all of the `--outfile` arguments to have more descriptive names instead of `runX.xml`.

Reply:
Great idea to put additional information in the file name. Could you also include the word "results" somewhere? Would help someone who has less experience with our smoke tests.